stock_cycle_count/models/stock_cycle_count_rule.py

```diff
@@ -0,0 +1,198 @@
+        string='Turnover Inventory Value Threshold')
+    currency_id = fields.Many2one(comodel_name='res.currency',
+                                  string='Currency',
+                                  compute=_compute_currency)
+    accuracy_threshold = fields.Float(string='Minimum Accuracy Threshold',
+                                      digits=(3, 2))
+    warehouse_ids = fields.Many2many(comodel_name='stock.warehouse',
+                                     relation='warehouse_cycle_count_rule_rel',
+                                     column1='rule_id',
+                                     column2='warehouse_id',
+                                     string='Applied in')
+
+    def compute_rule(self, locs):
+        if self.rule_type == 'periodic':
+            proposed_cycle_counts = self._compute_rule_periodic(locs)
+        elif self.rule_type == 'turnover':
+            proposed_cycle_counts = self._compute_rule_turnover(locs)
+        elif self.rule_type == 'accuracy':
+            proposed_cycle_counts = self._compute_rule_accuracy(locs)
+        return proposed_cycle_counts
+
+    @api.model
+    def _propose_cycle_count(self, date, location):
+        cycle_count = {
+            'date': date.strftime('%Y-%m-%d %H:%M:%S'),
+            'location': location,
+            'rule_type': self
+        }
+        return cycle_count
+
+    @api.model
+    def _compute_rule_periodic(self, locs):
+        cycle_counts = []
+        for loc in locs:
+            last_inventories = self.env['stock.inventory'].search([
+                ('location_id', '=', loc.id),
+                ('state', 'in', ['confirm', 'done', 'draft'])]).mapped('date')
+            if last_inventories:
+                latest_inventory = sorted(last_inventories, reverse=True)[0]
+                try:
+                    period = self.periodic_count_period / \
+                        self.periodic_qty_per_period
+                    next_date = datetime.strptime(
+                        latest_inventory, '%Y-%m-%d %H:%M:%S') + timedelta(
+                        days=period)
+                except Exception as e:
+                    raise UserError(
+                        _('Error found determining the frequency of periodic '
+                          'cycle count rule. %s') % e.message)
+            else:
+                next_date = datetime.today()
+            cycle_count = self._propose_cycle_count(next_date, loc)
+            cycle_counts.append(cycle_count)
+        return cycle_counts
+
+    @api.model
+    def _get_turnover_moves(self, location, date):
+        moves = self.env['stock.move'].search([
+            '|', ('location_id', '=', location.id),
+            ('location_dest_id', '=', location.id),
+            ('date', '>', date),
+            ('state', '=', 'done')])
+        return moves
+
+    @api.model
+    def _compute_turnover(self, move):
+        turnover = move.product_uom_qty * move.product_id.standard_price
```

Comment:
I'm attending this now and I found that the method you refer to (price_compute) is an odoo v10 method. What do you think about keeping it like this now and to add that change in a future migration to v10?